spec/requests/api/v1/documents_spec.rb

```diff
@@ -2,55 +2,60 @@
 require 'uri'
 
 describe API::V1::Documents do
-  let(:id) { "some really!weird@id.name" }
-  let(:valid_params) do
-    {
-      "document_id" => id,
-      "title" => "my title",
-      "path" => "http://www.gov.gov/goo.html",
-      "description" => "my desc",
-      "promote" => true,
-      "language" => 'hy',
-      "content" => "my content",
-      "tags" => "Foo, Bar blat",
-    }
+  let(:id) { 'some really!weird@id.name' }
+
+  let(:valid_session) do
+    credentials = ActionController::HttpAuthentication::Basic.encode_credentials 'test_index', 'test_key'
+    { HTTP_AUTHORIZATION: credentials }
   end
 
+  let(:allow_updates) { true }
+
+  let(:maintenance_message) { nil }
+
   before(:all) do
     yaml = YAML.load_file("#{Rails.root}/config/secrets.yml")
     env_secrets = yaml[Rails.env]
     credentials = ActionController::HttpAuthentication::Basic.encode_credentials env_secrets['admin_user'], env_secrets['admin_password']
-    valid_collection_session = { 'HTTP_AUTHORIZATION' => credentials }
-    valid_collection_params = { "handle" => "test_index", "token" => "test_key" }
-    post "/api/v1/collections", params: valid_collection_params, headers: valid_collection_session
+    valid_collection_session = { HTTP_AUTHORIZATION: credentials }
+    valid_collection_params = { handle: 'test_index', token: 'test_key' }
+    post '/api/v1/collections', params: valid_collection_params, headers: valid_collection_session
     Document.index_name = Document.index_namespace('test_index')
   end
 
-  let(:valid_session) do
-    credentials = ActionController::HttpAuthentication::Basic.encode_credentials "test_index", "test_key"
-    { 'HTTP_AUTHORIZATION' => credentials }
-  end
-
-  let(:allow_updates) { true }
-  let(:maintenance_message) { nil }
   before do
     I14y::Application.config.updates_allowed = allow_updates
     I14y::Application.config.maintenance_message = maintenance_message
   end
+
   after do
     I14y::Application.config.updates_allowed = true
   end
 
-  describe "POST /api/v1/documents" do
+  describe 'POST /api/v1/documents' do
+    let(:valid_params) do
+      { document_id: id,
+        title:       'my title',
+        path:        'http://www.gov.gov/goo.html',
+        description: 'my desc',
+        promote:     true,
+        language:    'hy',
+        content:     'my content',
+        tags:        'Foo, Bar blat' }
+    end
+
     context 'success case' do
       before do
         Elasticsearch::Persistence.client.delete_by_query index: Document.index_name, q: '*:*', conflicts: 'proceed'
```

Comment:
Can we remove all these `delete_by_query` calls now? I would assume `Document.refresh_index!` is doing the cleanup now?

Reply:
Ah - I misunderstood. I thought `Document.refresh_index!` was dropping & recreating the index.

Reply:
I'm still thinking that it would be cleaner, and possibly safer, to simply create the index before each spec, and delete if after, rather than using the `delete_by_query` calls. I like the approach they took in the Medium article you shared, where they tag the specs relying on ES and only run the index creation/deletion for those specs.